neard/src/main.rs

```diff
@@ -1,3 +1,7 @@
+#[cfg(not(feature = "memory_profiling"))]
+#[cfg(jemallocator)]
+extern crate jemallocator;
```

Comment:
I wanted to keep the behavior the same in case `memory_profiling` is not enabled as before the change.
You can take a look at `core/primitives/src/lib.rs`